src/c-writer.cc

```diff
@@ -712,9 +743,16 @@ void CWriter::Write(std::string_view s) {
   WriteData(s.data(), s.size());
 }
 
-void CWriter::Write(const LocalName& name) {
-  assert(local_sym_map_.count(name.name) == 1);
-  Write(local_sym_map_[name.name]);
+void CWriter::Write(const ParamName& name) {
+  std::string mangled = name.name + kParamSuffix;
+  assert(local_sym_map_.count(mangled) == 1);
+  Write(local_sym_map_[mangled]);
+}
+
+void CWriter::Write(const LabelName& name) {
+  std::string mangled = name.name + kLabelSuffix;
```

Comment:
Would it make the output more readable if this was a prefix such `l_` for labels and `p_` for params?    We can do that as part of your larger/followup change too of course.

Reply:
This doesn't affect the output -- it's only an internal tag to keep these names distinct in the symbol maps. The output is still (as close as possible to) the original name.